presto-docs/src/main/sphinx/connector/hive-storage-caching.rst

```diff
@@ -0,0 +1,195 @@
+
+**Reduced load on object storage**
+
+Every retrieved and cached object avoids repeated retrieval from the storage in
+subsequent queries. As a result the object storage system does not have to
+provide the object again and again.
+
+For example, if your query accesses 100MB of objects from the storage, the first
+time the query runs 100MB are downloaded and cached. Any following query uses
+these objects. If your users run another 100 queries accessing the same objects,
+your storage system does not have to do any significant work. Without caching it
+has to provide the same objects again and again, resulting in 10GB of total
+storage to serve.
+
+This reduced load on the object storage can also impact the sizing, and
+therefore the cost, of the object storage system.
+
+**Increased query performance**
+
+Caching can provide significant performance benefits, by avoiding the repeated
+network transfers and instead accessing copies of the objects from a local
+cache. Performance gains are more significant if the performance of directly
+accessing the object storage is low compared to accessing the cache.
+
+For example, if you access object storage in a different network, different data
+center or even different cloud-provider region query performance is slow. Adding
+caching using fast, local storage has a significant impact and makes your
+queries much faster.
+
+On the other hand, if you object storage is already running at very high
+performance for I/O and network access, and your local cache storage is at
+similar speeds, or even slower, performance benefits can be minimal.
+
+**Reduced query costs**
+
+A result of the reduced load on the object storage, mentioned earlier, is
+significantly reduced network traffic. Network traffic however is a considerable
+cost factor in an setup, specifically also when hosted in public cloud provider
+systems.
+
+Architecture
+------------
+
+Caching for distributed storage provides a read-through cache. After first
+retrieval from storage by any query, objects are cached in the local cache
+storage on the workers. Objects are cached on local storage of each worker and
+managed by a BookKeeper component. Workers can request cached objects from other
+workers to avoid requests from the object storage. The cache chunks are 1MB in
+size and are well suited for ORC or Parquet format objects.
+
+Configuration
+-------------
+
+The caching feature is part of the :doc:`Hive connector </connector/hive>` and
+can be activated in the catalog properties file:
+
+.. code-block:: none
+
+    connector.name=hive-hadoop2
+    hive.cache.enabled=true
+    hive.cache.location=/opt/hive-cache
+
+The cache operates on the coordinator and all workers accessing the object
+storage. The used networking ports for the managing BookKeeper and the data
+transfer, by default 8898 and 8899, need to be available.
+
+To use caching on multiple catalogs, you need to configure different caching
```

Comment:
I would skip `different` word. For ports it's obvious that they need to be different. For directories we could mention that more than single caching directory is preferred for performance reasons.